spec/controllers/two_factor_authentication/otp_verification_controller_spec.rb

```diff
@@ -653,6 +659,41 @@
           it 'resets context to authentication' do
             expect(subject.user_session[:context]).to eq 'authentication'
           end
+
+          context 'with new device session value' do
+            before do
+              subject.user_session[:new_device] = false
+              post(
+                :create,
+                params: {
+                  code: subject.current_user.direct_otp,
+                  otp_delivery_preference: 'sms',
+                },
+              )
+            end
+            it 'tracks new device value' do
+              subject.user_session[:new_device] = false
+              parsed_phone = Phonelib.parse('+1 (703) 555-5555')
+              properties = {
+                success: true,
+                errors: nil,
+                context: 'confirmation',
+                multi_factor_auth_method: 'sms',
+                multi_factor_auth_method_created_at: nil,
+                new_device: nil,
```

Comment:
* Do we need `subject.user_session[:new_device] = false` twice between `before` and the `it` tests, or should we only have it once?
* Wouldn't we expect `new_device: false` here instead of `new_device: nil` ?

Reply:
Fixed those errors. Should be working correctly now.